msu/systems/serialization/serialization_system.nut

```diff
@@ -14,4 +16,53 @@
 		this.Mods.push(_mod);
 		_mod.Serialization = ::MSU.Class.SerializationModAddon(_mod);
 	}
+
+	function flagSerialize( _mod, _id, _object, _flags = null )
+	{
+		if (_flags == null) _flags = ::World.Flags;
+		local outEmulator = ::MSU.Class.SerializationEmulator(_mod, _id);
+		if (::MSU.isBBObject(_object))
```

Comment:
So, the assumption is that if you are passing a BBObject as `_object` then it must have an onSerialize function. Correct? Because we don't want to serialize BBObjects as tables.

Reply:
Yes, specifically the 2nd part, but basically 100% exactly what you said.